server.js

```diff
@@ -13,8 +13,64 @@ app.get("/", function (request, response) {
   response.send("Hotel booking server.  Ask for /bookings, etc.");
 });
 
-// TODO add your routes and helper functions here
+app.get("/bookings", function (req, res) {
+  res.send(bookings);
+});
+
+app.get("/bookings/search", function (req, res) {
+  if (req.query.date) {
+    const date = dateToNumber(req.query.date);
+    res.send(bookings.filter((booking) => dateToNumber(booking.checkInDate) <= date && dateToNumber(booking.checkOutDate) >= date));
```

Comment:
would be better to extract the filtration function, it would be more readable

Reply:
yes I will do that thanks